### ISSUE TITLE
Passing Y vector to Convolution layer

### ISSUE THREAD

**Issue by Timopheym** [dmlc/mxnet#1010]
Thanks for wonderful library,
i am trying to run lenet from python example with Kaggle csv train set.
I'v done reshape for x Matrix 
```python
train_X = train_X.reshape((len(train_X), 28, 28))
```
And pass the data
```
    model.fit(X= train_X, y = train_Y)
```
But when i run it - i've got error
```
  File "/Users/timopheym/Desktop/Projects/data_analysis/bin/mxnet/example/image-classification/train_model_poc.py", line 67, in fit
    y                  = train_Y,
...

src/operator/./convolution-inl.h:301: Check failed: (dshape.ndim()) == (4) Input data should be 4D in batch-num_filter-y-x
```

Why should train_Y be 4D? This is only labels vector... 
I've read r tutorial - but there no information about that, so there are no doc's about Convolution layer. Can you help me please? 
Here is the full code https://gist.github.com/Timopheym/3f3bd380f61876f7e7a5

**Comment by tornadomeet** [comment]
the error message here is not about the train_Y, but about the train_X; because during convolution computation, it does't need the label(only the output layer will used label).
in```train_X = train_X.reshape((len(train_X), 28, 28))```, train_X is only three dim right now, 
you may need to change ```train_X = train_X.reshape((len(train_X), 1, 28, 28))```

**Comment by Timopheym** [comment]
@tornadomeet Sorry, my fault. 
Thanks a lot, that helped.
